test/web/mock.js

```diff
@@ -1,83 +1,103 @@
-    if(headers) {
-      if(headers.json) {
-        headers['content-type'] = 'application/json';
-        delete headers.json;
-      }
-    } else {
-      headers = {};
+      // this creates the same invocationSigner for each test.
+      this.invocationSigner = new MockInvoker();
+      // create KAK and HMAC keys for creating edvs
+      // this creates the same keyAgreementKey for each test.
+      this.keys.keyAgreementKey = new MockKak();
+      // the creates the same hmac for each test.
+      this.keys.hmac = await MockHmac.create();
+      // only store the KaK in the recipients' keyStorage.
+      this.keyStorage.set(
+        this.keys.keyAgreementKey.id, this.keys.keyAgreementKey);
+      this.keyResolver = ({id}) => {
+        const key = this.keyStorage.get(id);
+        if(key) {
+          return key;
+        }
+        throw new Error(`Key ${id} not found`);
+      };
+      this.documentLoader = async url => {
+        if(url.startsWith('did:key:')) {
+          const id = url.replace('did:key:');
+          return {
+            contextUrl: null,
+            documentUrl: url,
+            document: this.keyResolver({id})
+          };
+        }
+        if(url === 'https://www.w3.org/ns/did/v1') {
+          return {
+            contextUrl: null,
+            documentUrl: url,
+            document: didContext.contexts.get('https://w3id.org/did/v0.11')
+          };
+        }
+      };
     }
-    return headers;
-  };
-  server.prepareBody = function prepareBody(body, headers) {
-    if(headers && headers['content-type'] === 'application/json') {
-      return (body && typeof body !== 'string') ?
-        JSON.stringify(body) : '{"message": "mock server error"}';
+  }
+  async createEdv({controller, referenceId, keyResolver} = {}) {
+    const {keyAgreementKey, hmac} = this.keys;
+    let config = {
+      sequence: 0,
+      controller: controller || this.invocationSigner.id,
+      keyAgreementKey: {id: keyAgreementKey.id, type: keyAgreementKey.type},
+      hmac: {id: hmac.id, type: hmac.type}
+    };
+    if(referenceId) {
+      config.referenceId = referenceId;
     }
-    return body;
-  };
-
-  // mock backend for KMS
-  mock.kms = new MockKmsService({server});
-
-  // mock edv storage
-  mock.edvStorage = new MockStorage({server, controller: accountId});
-
-  // only init keys once
-  if(!mock.keys) {
-    // create mock keys
-    mock.keys = {};
-
-    // // account master key for using KMS
-    // const secret = 'bcrypt of password';
-    // const kmsService = new KmsService();
-    // mock.keys.master = await AccountMasterKey.fromSecret(
-    //   {secret, accountId, kmsService, kmsPlugin: 'mock'});
-    //
-    // // create KEK and HMAC keys for creating edvs
-    // mock.keys.kek = await mock.keys.master.generateKey({type: 'kek'});
-    // mock.keys.hmac = await mock.keys.master.generateKey({type: 'hmac'});
+    config = await EdvClient.createEdv(
+      {config, url: 'http://localhost:9876/edvs'});
```

Comment:
I recognize this port as a karma proxy.  The `http` protocol being used here concerns me.  Is this how it was before?

Reply:
https://github.com/digitalbazaar/edv-client/blob/main/tests/mock.js#L69 This is how it is in edv-client and the mock.js stuff from there